src/routes/pois/components/Pois.js

```diff
@@ -0,0 +1,146 @@
+// @flow
+
+import * as React from 'react'
+import { View } from 'react-native'
+import type { TFunction } from 'react-i18next'
+import {
+  CATEGORIES_FEEDBACK_TYPE,
+  CityModel,
+  CONTENT_FEEDBACK_CATEGORY,
+  PoiModel,
+  EVENTS_FEEDBACK_TYPE,
+  PAGE_FEEDBACK_TYPE,
+  TECHNICAL_FEEDBACK_CATEGORY
+} from '@integreat-app/integreat-api-client'
+import Page from '../../../modules/common/components/Page'
+import ContentNotFoundError from '../../../modules/error/ContentNotFoundError'
+import PageDetail from '../../../modules/common/components/PageDetail'
+import List from '../../../modules/common/components/List'
+import Caption from '../../../modules/common/components/Caption'
+import Failure from '../../../modules/error/components/Failure'
+import type { ThemeType } from '../../../modules/theme/constants/theme'
+import type { LanguageResourceCacheStateType } from '../../../modules/app/StateType'
+import type { NavigationScreenProp } from 'react-navigation'
+import type { NavigateToEventParamsType } from '../../../modules/app/createNavigateToEvent'
+import type { NavigateToIntegreatUrlParamsType } from '../../../modules/app/createNavigateToIntegreatUrl'
+import FeedbackVariant from '../../feedback/FeedbackVariant'
+import SiteHelpfulBox from '../../../modules/common/components/SiteHelpfulBox'
+import SpaceBetween from '../../../modules/common/components/SpaceBetween'
+import type {
+  FeedbackCategoryType,
+  FeedbackType
+} from '@integreat-app/integreat-api-client/endpoints/createFeedbackEndpoint'
+import ErrorCodes from '../../../modules/error/ErrorCodes'
+import PoiListItem from './PoiListItem'
+
+export type PropsType = {|
+  path: ?string,
+  pois: Array<PoiModel>,
+  cities: Array<CityModel>,
+  cityCode: string,
+  language: string,
+  resourceCache: LanguageResourceCacheStateType,
+  theme: ThemeType,
+  t: TFunction,
+  navigation: NavigationScreenProp<*>,
+  navigateToPoi: NavigateToEventParamsType => void,
+  navigateToIntegreatUrl: NavigateToIntegreatUrlParamsType => void
+|}
+
+/**
+ * Displays a list of events or a single event, matching the route /<location>/<language>/events(/<id>)
+ */
+class Pois extends React.Component<PropsType> {
+  navigateToPoi = (cityCode: string, language: string, path: string) => () => {
+    this.props.navigateToPoi({ cityCode, language, path })
+  }
+
+  renderPoiListItem = (cityCode: string, language: string) => (poi: PoiModel) => {
+    const { theme } = this.props
+    return <PoiListItem key={poi.path}
+                          poi={poi}
+                          language={language}
+                          theme={theme}
+                          navigateToPoi={this.navigateToPoi(cityCode, language, poi.path)} />

```

Comment:
```suggestion
                          navigateToPoi={this.props.navigateToPoi({ cityCode, language, path: poi.path })} />

```

Reply:
@maxammann  i tried to apply both of your changes to `Pois.js` , but they are not working this way.
I also tried to fix it, but my typescript skills are maybe not enough.
I get the error that the return type is not correct !?
Should we leave it the old way? Or can you provide me a proper fix? 🧐

Reply:
lets leave it the old way :)